### PR TITLE
feat(app): App add reset option for run history

### PR DESCRIPTION
**Pull request by koji**
# Overview
This will close the rest of #8792 which is runsHistory reset option and runsHistory log download
#10395 has been merged into `edge` then the app can offer runsHistory log reset.

# Changelog
- Add runs history download button
- Add checkbox for runs history

# Review requests
- runsHistory option is added recently to the robot-server so that this PR requires the latest robot-server to see the option.
 

# Risk assessment
low

### CODE REVIEW

File: app/src/assets/localization/en/device_settings.json [hunk 1]
```diff
@@ -115,12 +115,12 @@
   "factory_reset_slideout_title": "Factory Reset",
   "factory_reset_slideout_description": "Select the robot data to clear.",
   "factory_reset_slideout_warning_message": "Factory resets cannot be undone",
-  "factory_reset_slideout_calibration_title": "Robot calibration data",
-  "factory_reset_slideout_calibration_description": "Resetting Deck and/or Tip Length Calibration data will also clear Pipette Offset Calibration data.",
-  "factory_reset_slideout_protocol_run_history_title": "Protocol Run History",
-  "factory_reset_slideout_boot_scripts_title": "Boot Scripts",
-  "factory_reset_slideout_data_clear_button": "Clear data and restart robot",
-  "factory_reset_slideout_download_logs_link": "Download",
+  "reset_option_calibration": "Robot calibration data",
```
Comment (koji) [inline app/src/assets/localization/en/device_settings.json:121]: I removed `factory_reset_slideout` since they are unique without it.

**Review by smb2268 — APPROVED** [review]
tested this out and it looks good!

**Comment by koji** [comment]
~~Before merging the following should be solved~~
~~#10477~~
#10477 doesn't block this because it is about `runsHistory` which is solved by this PR

File: app/src/organisms/Devices/RobotSettings/AdvancedTab/AdvancedTabSlideouts/__tests__/FactoryResetSlideout.test.tsx [hunk 2]
```diff
@@ -30,6 +30,11 @@ const mockResetConfigOptions = [
     name: 'Calibration Bar',
     description: 'Calibration bar description',
   },
+  // {
```
Comment (b-cooper) [inline app/src/organisms/Devices/RobotSettings/AdvancedTab/AdvancedTabSlideouts/__tests__/FactoryResetSlideout.test.tsx]: is this intentionally commented out?

Comment (koji) [reply app/src/organisms/Devices/RobotSettings/AdvancedTab/AdvancedTabSlideouts/__tests__/FactoryResetSlideout.test.tsx]: I forgot to update the test case.
Now the `//`s are removed

File: app/src/organisms/Devices/RobotSettings/RobotSettingsAdvanced.tsx [hunk 3]
```diff
@@ -95,6 +98,7 @@ export function RobotSettingsAdvanced({
     if (options != null) setResetOptions(options)
     setShowFactoryResetModal(true)
     setIsRobotConnected(isConnected ?? false)
+    history.push(`/devices/`)
```
Comment (koji) [inline app/src/organisms/Devices/RobotSettings/RobotSettingsAdvanced.tsx]: After reset, redirect to the device list page

**Pull request merged**